### ISSUE TITLE
Newly imposed 63-character limit consequences.

### ISSUE THREAD

**Issue by dmzoneill** [RedHatInsights/bonfire#288]
Our config in app-interface which is used by Jenkins/GitLab webhooks etc, burns 49 characters before we add anything to the label string.

For example, automation-analytics-automation-analytics-backend-cypress-pr-check-2 is 69 characters.
6 characters over the limit imposed.

It's not feasible to just change gl_group/gl_project as these are used in Jenkins REST callbacks/pr decoration etc.

I propose that bonfire not use these variables in the name space reservation, as it breaks the 63 character.
We need this changed as quickly as possible.

```config:
- project:
    name: automation-analytics-ci
    label: insights
    node: insights
    gl_group: automation-analytics
    gl_project: automation-analytics-backend
    quay_org: cloudservices
```

10:55:07 pr-681-latest: digest: sha256:314110fb9e19bd96b402a47f2be0f823005d3adc5cd2a31c4eaffe8d9ce8caed size: 2228
10:55:07 ++ set +x
10:55:07 Mon Feb 13 10:55:07 UTC 2023 *** To start deployment
10:55:07 /var/lib/jenkins/workspace/automation-analytics-automation-analytics-backend-cypress-pr-check/.bonfire/cicd/bin/oc_wrapper
10:55:07 2023-02-13 10:55:07 [    INFO] [          MainThread] checking pypi for latest release...
10:55:07 2023-02-13 10:55:07 [    INFO] [          MainThread] up-to-date!
10:55:07 2023-02-13 10:55:07 [ WARNING] [          MainThread] split() requires a non-empty pattern match.
10:55:08 2023-02-13 10:55:08 [    INFO] [          MainThread] Checking for existing reservations for 'automation-analytics-automation-analytics-backend-cypress-pr-check-2'
10:55:08 2023-02-13 10:55:08 [    INFO] [          MainThread] checking for available namespaces to reserve...
10:55:08 2023-02-13 10:55:08 [    INFO] [          MainThread] pool size limit is defined as 0 in 'default' pool
10:55:08 2023-02-13 10:55:08 [    INFO] [          MainThread] processing namespace reservation
10:55:08 2023-02-13 10:55:08 [    INFO] [          MainThread] running (pid 14711): oc apply -f - 
10:55:08 2023-02-13 10:55:08 [    INFO] [           pid-14711]  |stderr| The NamespaceReservation "bonfire-reservation-719bf53e" is invalid: metadata.labels: Invalid value: "automation-analytics-automation-analytics-backend-cypress-pr-check-2": must be no more than 63 characters

**Comment by bsquizz** [comment]
This is probably because `JOB_NAME` is unexpectedly long, and bootstrap.sh sets the NS requester using the job name here: https://github.com/RedHatInsights/bonfire/blob/master/cicd/bootstrap.sh#L22

As a quick work-around, set the `BONFIRE_NS_REQUESTER` env var to be something shorter at some point after 'bootstrap.sh' is invoked in your pr check shell script.

Suggestion: `export BONFIRE_NS_REQUESTER=aa-cypress-pr-check-${BUILD_NUMBER}`

**Comment by dmzoneill** [comment]
I've hacked at the bits I could, and got it below the 63 character limit.
automation-analytics-automation-analytics-backend-cypress (57)

https://ci.int.devshift.net/job/automation-analytics-automation-analytics-backend-cypress/4/

**Comment by bsquizz** [comment]
What was the work-around you added?

**Comment by mshriver** [comment]
This line prevents setting `BONFIRE_NS_REQUESTER` from working, this issue is still active. 

https://github.com/RedHatInsights/bonfire/blob/master/cicd/deploy_ephemeral_env.sh#L19

Although its set in bootstrap, its set again here.

**Comment by bsquizz** [comment]
Good point. It doesn't need to be set twice. https://github.com/RedHatInsights/bonfire/pull/307

**Comment by dmzoneill** [comment]
looks like @bsquizz's PR #307 addresses the double-setting issue that @mshriver pointed out. the root cause seems to be that setting BONFIRE_NS_REQUESTER in bootstrap.sh gets overwritten by deploy_ephemeral_env.sh. once #307 merges, the workaround should work correctly.
